src/ast/nodes/MemberExpression.ts

```diff
@@ -218,20 +232,17 @@ export default class MemberExpression
 		if (this.isUndefined) {
 			return undefined;
 		}
-		if (this.propertyKey !== UnknownKey && path.length < MAX_PATH_DEPTH) {
+		const propertyKey = this.getDynamicPropertyKey();
+		if (propertyKey !== UnknownKey && path.length < MAX_PATH_DEPTH) {
 			this.expressionsToBeDeoptimized.push(origin);
```

Comment:
I tried `4.29.0-0` out and got `TypeError: Cannot add property 0, object is not extensible` error at this line. 
If I replace this line to `this.expressionsToBeDeoptimized = [...EMPTY_ARRAY]`, the error doesn't happen.
https://github.com/rollup/rollup/pull/5737/files#diff-7421c7a9da2cb68b551f48a8bd91f00c376f93e0a07b86f4563b1aab964c64a3R198

The way to reproduce is to clone https://github.com/vitejs/vite and upgrade rollup in `packages/vite/package.json` and run `pnpm build` in `packages/vite`. Let me know if I should trim it down.

Reply:
Great catch, will have a deeper look by tomorrow. Maybe we just need to use a fresh empty array rather than the shared and (for good reasons) frozen empty array, or maybe there is a wrong assumption somewhere.

Reply:
I see. I changed the logic im MemberExpressions to not deoptimize fully but only to its statically resolvable value if that is knowable, but there are places in the logic that still assume we always deoptimize to "unknown". Will likely draw up a fix tomorrow. In the meantime, here is a minimal reproduction that I managed to distill from the Vite code
https://rollup-eztxif0az-rollup-js.vercel.app/repl/?pr=5737&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmZ1bmN0aW9uJTIwdGVzdCh4KSUyMCU3QiU1Q24lMjAlMjBpZiUyMCh4JTNGLnkueiklMjAlN0IlNUNuJTIwJTIwJTIwJTIwY29uc29sZS5sb2coJ29rJyklNUNuJTIwJTIwJTdEJTVDbiU3RCU1Q24lNUNudGVzdCglN0J5JTNBJTIwdHJ1ZSU3RCklM0IlNUNudGVzdCglN0J5JTNBJTIwZmFsc2UlN0QpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlN0QlN0Q=

Reply:
Thanks! The build and tests in the vite repo now passes 👍

Reply:
May I ask for a new pre-release? I'd like to try it out on vite-ecosystem-ci. I tried to run it with a patch but it didn't work easily.

Reply:
Should be there in 20-30 minutes

Reply:
analogjs, nuxt, qwik, sveltekit, vitest didn't pass the ecosystem-ci (but was passing with previous rollup versions).
I digged down nuxt and made a minimal reproduction:
https://rollup-eztxif0az-rollup-js.vercel.app/repl/?pr=5737&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwa2V5cyUyMCUzRCUyMCU3QiU1Q24lMjAlMjBuYW1lJTNBJTIwJ25hbWUnJTJDJTVDbiUyMCUyMG9yZGVyJTNBJTIwJ29yZGVyJyUyQyU1Q24lMjAlMjBlbmZvcmNlJTNBJTIwJ2VuZm9yY2UnJTJDJTVDbiUyMCUyMGRlcGVuZHNPbiUzQSUyMCdkZXBlbmRzT24nJTJDJTVDbiU3RCU1Q25mdW5jdGlvbiUyMGlzTWV0YWRhdGFLZXklMjAoa2V5KSUyMCU3QiU1Q24lMjAlMjByZXR1cm4lMjBrZXklMjBpbiUyMGtleXMlNUNuJTdEJTVDbiU1Q25jb25zb2xlLmxvZyhpc01ldGFkYXRhS2V5KCduYW1lJykpJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlN0QlN0Q=

I'll take a look at others whether it's a bug in rollup.

Reply:
It seems Vitest's failure is also caused by a bug.
Steps to reproduce the vitest's error:

1. clone https://github.com/vitest-dev/vitest
1. change this line https://github.com/vitest-dev/vitest/blob/5ba49475dc1efe10a4d43443f70c683b9fde8720/package.json#L75 to `"rollup": "4.29.0-1"`
1. run `pnpm i && pnpm build`
1. at `../../packages/vitest/dist/chunks/resolveConfig.DCdQOwtR.js:6791:33`, `getConfig` has no argument passed. but it should have `project` (https://github.com/vitest-dev/vitest/blob/5ba49475dc1efe10a4d43443f70c683b9fde8720/packages/vitest/src/node/pools/forks.ts#L217).

Reply:
I made  a minimal reproduction for qwik: https://rollup-eztxif0az-rollup-js.vercel.app/repl/?pr=5737&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmNvbnN0JTIwc2FmZUNhbGwlMjAlM0QlMjAoY2FsbCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwdHJ5JTIwJTdCJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMGNhbGwoKSUzQiU1Q24lMjAlMjAlN0QlMjBjYXRjaCUyMChlKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB0aHJvdyUyMGUlNUNuJTIwJTIwJTdEJTVDbiU3RCUzQiU1Q24lNUNuY29uc3QlMjBmb28lMjAlM0QlMjAxJTVDbmNvbnN0JTIwb3B0cyUyMCUzRCUyMCU3QiUyMGZvbyUyMCU3RCU1Q25zYWZlQ2FsbCgoKSUyMCUzRCUzRSUyMGNvbnNvbGUubG9nKG9wdHMpKSUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTdEJTdE

Reply:
Sveltekit failed locally with and without upgrading rollup, so probably that one is not related.

Reply:
I resolved the two issues you created the two reproductions for in 4.29.0. I think this also fixed Vitest. Will need to have a look at analog tomorrow.

Reply:
And thank you so much, ecosystem CI is really pure gold here. In the future, can I trigger this some way from my side for "risky" releases, e.g. by creating a draft PR against the Vite repo with a beta version?

Reply:
It seems to me that analog runs with 4.29.0-2 with similar performance as before.

Reply:
Thanks for the fixes. It seems analog and storybook is failing now (others are passing 🎉). I'll take a look tomorrow.

> In the future, can I trigger this some way from my side for "risky" releases, e.g. by creating a draft PR against the Vite repo with a beta version?

Yes 👍 I'll make a PR tomorrow so that you can run it.

Reply:
Made a reproduction for storybook (probably analog has the same root cause).
https://stackblitz.com/edit/rollup-repro-cqidpdk2?file=package.json,src%2Fmain.js,src%2Ffoo.js,rollup.config.js
(need to be downloaded and run locally as it doesn't work in stackblitz)

Also I found a related regression. There was a case that the code was removed before but now not removed ([4.29.1 REPL](https://rollupjs.org/repl/?version=4.29.1&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGbW9kdWxlXzEuanMnJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJ2YXIlMjBldmVudHMlMjAlM0QlMjAlMkYqJTIwJTQwX19QVVJFX18lMjAqJTJGJTIwKChldmVudHMyKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjBldmVudHMyJTVCJTVDJTIyQ0hBTk5FTF9XU19ESVNDT05ORUNUJTVDJTIyJTVEJTIwJTNEJTIwJTVDJTIyY2hhbm5lbFdTRGlzY29ubmVjdCU1QyUyMiUzQiU1Q24lMjAlMjBldmVudHMyJTVCJTVDJTIyQ0hBTk5FTF9DUkVBVEVEJTVDJTIyJTVEJTIwJTNEJTIwJTVDJTIyY2hhbm5lbENyZWF0ZWQlNUMlMjIlM0IlNUNuJTIwJTIwcmV0dXJuJTIwZXZlbnRzMiUzQiU1Q24lN0QpKGV2ZW50cyUyMCU3QyU3QyUyMCU3QiU3RCklM0IlNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGV2ZW50cyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMCU3QiU1Q24lMjAlMjBDSEFOTkVMX1dTX0RJU0NPTk5FQ1QlMkMlNUNuJTIwJTIwQ0hBTk5FTF9DUkVBVEVEJTVDbiU3RCUyMCUzRCUyMGV2ZW50cyUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJtb2R1bGVfMS5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlN0QlN0Q=), 
[4.30.1 REPL](https://rollup-eztxif0az-rollup-js.vercel.app/repl/?pr=5737&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCcuJTJGbW9kdWxlXzEuanMnJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJ2YXIlMjBldmVudHMlMjAlM0QlMjAlMkYqJTIwJTQwX19QVVJFX18lMjAqJTJGJTIwKChldmVudHMyKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjBldmVudHMyJTVCJTVDJTIyQ0hBTk5FTF9XU19ESVNDT05ORUNUJTVDJTIyJTVEJTIwJTNEJTIwJTVDJTIyY2hhbm5lbFdTRGlzY29ubmVjdCU1QyUyMiUzQiU1Q24lMjAlMjBldmVudHMyJTVCJTVDJTIyQ0hBTk5FTF9DUkVBVEVEJTVDJTIyJTVEJTIwJTNEJTIwJTVDJTIyY2hhbm5lbENyZWF0ZWQlNUMlMjIlM0IlNUNuJTIwJTIwcmV0dXJuJTIwZXZlbnRzMiUzQiU1Q24lN0QpKGV2ZW50cyUyMCU3QyU3QyUyMCU3QiU3RCklM0IlNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGV2ZW50cyUzQiU1Q25leHBvcnQlMjBjb25zdCUyMCU3QiU1Q24lMjAlMjBDSEFOTkVMX1dTX0RJU0NPTk5FQ1QlMkMlNUNuJTIwJTIwQ0hBTk5FTF9DUkVBVEVEJTVDbiU3RCUyMCUzRCUyMGV2ZW50cyUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSUyQyUyMm5hbWUlMjIlM0ElMjJtb2R1bGVfMS5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlN0QlN0Q=)).

Reply:
> > In the future, can I trigger this some way from my side for "risky" releases, e.g. by creating a draft PR against the Vite repo with a beta version?
> 
> Yes 👍 I'll make a PR tomorrow so that you can run it.

This PR has been merged. You should be able to run it by the following steps:

1. Make a draft PR against the Vite repo with the new rollup version.
2. Comment `/ecosystem-ci run` on that PR.

Let me know if it does not work for have any questions about the results of ecosystem-ci. 🙂